### PR TITLE
Update on infrastructure

### PR DESCRIPTION
**Pull request by mike4741**
I only add method the makes a call to the API  using Category and Type  , ones we are   clear with this two we can add more in the future .
 for this purpose I add one method in the event Item controller  that filter using  category/type  or both 

        [HttpGet("[action]/Catagory/{catagoryId}/Type{typeId}")]

### CODE REVIEW

File: EventCatalogAPI/Controllers/EventItemsController.cs [hunk 1]
```diff
@@ -208,9 +209,46 @@ private List<EventItem> ChangeImageUrl(List<EventItem> items)
             return items;
         }
 
+        [HttpGet("[action]/Catagory/{catagoryId}/Type{typeId}")]
```
Comment (misurac) [inline EventCatalogAPI/Controllers/EventItemsController.cs:212]: I like the idea of this addition - have you tested it yet? Let's test before we merge.

File: EventCatalogAPI/Controllers/EventItemsController.cs [hunk 2]
```diff
@@ -208,9 +209,46 @@ private List<EventItem> ChangeImageUrl(List<EventItem> items)
...
+        public async Task<IActionResult> Items(
+               [FromQuery] int? catagoryId,
+               [FromQuery] int? typeId,
+               [FromQuery] int pageIndx = 0,
```
Comment (misurac) [inline EventCatalogAPI/Controllers/EventItemsController.cs:216]: I think this should change from pageIndx to pageIndex. We might run into issues if we leave it

**Comment by misurac** [comment]
Everything in the Interface folder looks ready to merge. Left a couple comments on the API

File: EventCatalogAPI/Controllers/EventItemsController.cs (hunk 2 above)
Comment (mike4741) [reply EventCatalogAPI/Controllers/EventItemsController.cs:216]: ok  I will edit that

File: EventCatalogAPI/Controllers/EventItemsController.cs (hunk 1 above)
Comment (mike4741) [reply EventCatalogAPI/Controllers/EventItemsController.cs:212]: I tested it  and it works for me  but with some adjustment   ,  I push the update   made  based on the comment

**Comment by LalithaNadimpalli** [comment]
Can you redo this on the latest merge..

**Pull request closed without merging**